pkg/scheduler/backend/api_dispatcher/call_controller.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apidispatcher
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	fwk "k8s.io/kube-scheduler/framework"
+	"k8s.io/utils/buffer"
+)
+
+// queuedAPICall contains an API call and other fields needed for its processing by the callController.
+type queuedAPICall struct {
+	fwk.APICall
+	// onFinish is a channel to which the result of the API call's execution is sent in a non-blocking way.
+	// sendOnFinish helper should always be used instead of a direct channel.
+	onFinish chan<- error
+	// callID provides a unique identity for this enqueued call.
+	// It is used during finalization to differentiate this instance from a newer one that may have replaced it
+	// while this call was in-flight.
+	callID int
+}
+
+// sendOnFinish performs a non-blocking send of an error to the onFinish channel.
+// This method shoud be used instead of direct operation on the channel.
```

Comment:
```suggestion
// This method should be used instead of direct operation on the channel.
```
typo: "shoud" → "should"